libyara/proc.c

```diff
@@ -386,6 +387,22 @@ int _yr_process_attach(
     return ERROR_COULD_NOT_ATTACH_TO_PROCESS;
   }
 
+  status = 0;
+  if (waitpid(-1, &status, 0) == -1)
```

Comment:
You are correct, @hillu.

My reading of yara.c indicates that YARA itself doesn't use multiple threads for memory scanning, but anyone using libyara are free to do so. I've updated the patch to use waitpid(pid,...) as you've suggested.